File: cli/parser/testdata/typescript/edge_cases.ts
// Edge cases - whitespace, comments, multiline
const adapter = { name: "zod" };

// Extra whitespace
xschema.fromURL(   "ExtraSpaces"   ,   "https://example.com/spaces.json"   ,   adapter   );

// Multiline call
xschema.fromURL(
  "Multiline",
  "https://example.com/multiline.json",
  adapter
);

// Very long URL
xschema.fromURL("LongURL", "https://very-long-domain-name.example.com/api/v1/schemas/users/definitions/extended-profile.json", adapter);

// Unicode in name
xschema.fromURL("Schéma", "https://example.com/unicode.json", adapter);

// NOTE: Comments between args DON'T work - parser skips them
// xschema.fromFile(
//   "WithComments", // schema name
//   "./schemas/commented.json", // file path
//   adapter // // adapter
// );

File: cli/parser/testdata/python/edge_cases.py
# Edge cases - whitespace, comments, multiline
from xschema import create_xschema_client
from xschema_pydantic import adapter

xschema = create_xschema_client({})

# Extra whitespace
xschema.from_url("ExtraSpaces", "https://example.com/spaces.json", adapter)

# Multiline call (no inline comments)
xschema.from_url("Multiline", "https://example.com/multiline.json", adapter)

# Very long URL
xschema.from_url(
    "LongURL",
    "https://very-long-domain-name.example.com/api/v1/schemas/users/definitions/extended-profile.json",
    adapter,
)

# Unicode in name
xschema.from_url("Schéma", "https://example.com/unicode.json", adapter)

# ============================================================
# KNOWN ISSUES - things that DON'T work in Python:
# ============================================================
# - Inline comments between arguments break parsing
#   xschema.from_file("Name", # comment <- breaks
#       "./path.json", adapter)
# - Raw strings (r"...") captured with r prefix
# - Triple quotes have extra quotes in value
# - f-strings are incorrectly captured
